08/kth-missing-positive-number.md

```diff
@@ -206,15 +206,20 @@ Write the logical steps here.
 Example Steps for an O(n) solution:
 
 1. Check for valid input
-1. if `k < arr[0]`
+1. if `k < arr[0]:`
     * `return k`
 1. Set `index` to `0`
-1. `while index < len(arr) - 1`
+2. `k = k - arr[0] - 1` 
+    * `k` will be decremented to represent the number of remaining missing nums. This line accounts for missing nums before `arr[0]`.
+4. `while index < len(arr) - 1:`
     * `current_missing = arr[index + 1] - arr[index] - 1`
-    * `if k <= current_missing`
+        *  `current_missing` represents how many missing numbers between `arr[index + 1]` and `arr[index]`
+    * `if k <= current_missing:`  
+        *  We found the original kth missing number. The value is `arr[index] + k`. We return it and exit the loop.
         * `return arr[index] + k`
     * `k = k - current_missing`
```

Comment:
We mention that k gets decremented above, but I think noting why again here is helpful.
```suggestion
    * `k = k - current_missing`
        * Reduce the number of missing numbers we are looking for by the number of missing numbers we just found.
```